_posts/2023-03-16-back-to-terraform.md

```diff
@@ -0,0 +1,226 @@
+
+## Additional complexity without the promised benefit: Terragrunt
+
+### Background
+
+When starting a new infrastructure project, it's important to choose the right tools for the job. But what happens when you inherit an existing project with infrastructure as code (IaC) already in place, using a tool like Terragrunt that your team is not familiar with? This is the situation that the team in this story found themselves in.
+
+The project they were working on was based on an existing infrastructure stack that had been built using Terragrunt, a popular wrapper around Terraform. While Terragrunt can be a powerful tool for managing complex infrastructure stacks, the new team found that it added unnecessary complexity and overhead to their workflow. They struggled to read and understand the existing Terragrunt code, which had been split into multiple Terraform modules in different git repositories and brought back together using a Terragrunt configuration repository.
+
+Adding to the complexity, the Terragrunt configuration was loading different versions of the Terraform modules in different environments, and the Terraform state was split into modules as well, with custom scripts to read and manipulate the state. All of these factors made it difficult for the new team to make progress on the project, and they began to question the decision to use Terragrunt in the first place.
+
+### Deciding to Return to Terraform
+
+After struggling to work with the complex and fragmented Terragrunt setup, the new infrastructure team wanted to find a simpler, more streamlined approach. They sought a setup with a low threshold, low maintenance, and easy to understand.
+
+As they dug deeper into the Terragrunt configuration, they found that even the company's architects had some doubts about its usefulness. So they began discussing alternatives, and eventually someone asked the question: "Why are we using Terragrunt?"
+
+The answer was surprising: the team wasn't even using any of Terragrunt's real advantages, and was instead dealing with unnecessary overhead. The configuration of all external modules was just a single key-value file, with no clear indication of which value was passed to which module.
+
+It became clear that continuing to use Terragrunt was simply pointless, and only adding to the team's frustration. They, together with the company's architects, made the decision to switch back to using Terraform directly, in order to simplify their workflow and make progress on the project.
+
+### Returning to terraform without breaking the existing infrastructure
+
+After making the decision to switch back to using Terraform directly, the team realized that a complete overhaul of their infrastructure setup would be necessary. Rather than throwing out all of the existing Terragrunt code, however, they decided to take a hybrid approach.
```

Comment:
```suggestion
After making the decision to switch back to using Terraform directly, the team realized that a complete overhaul of their infrastructure setup would be necessary. 
Rather than throwing out all of the existing Terragrunt code, however, they decided to take a hybrid approach.
```